src/ImageSharp/Formats/Jpeg/Components/Block8x8F.cs

```diff
@@ -596,5 +600,98 @@ private static void GuardBlockIndex(int idx)
             DebugGuard.MustBeLessThan(idx, Size, nameof(idx));
             DebugGuard.MustBeGreaterThanOrEqualTo(idx, 0, nameof(idx));
         }
+
+        /// <summary>
+        /// Transpose the block into the destination block.
+        /// </summary>
+        /// <param name="d">The destination block</param>
+        [MethodImpl(InliningOptions.ShortMethod)]
+        public void TransposeInto(ref Block8x8F d)
+        {
+#if SUPPORTS_RUNTIME_INTRINSICS
+            if (Avx.IsSupported)
+            {
+                this.TransposeIntoAvx(ref d);
+            }
+            else
+#endif
+            {
+                this.TransposeIntoFallback(ref d);
+            }
+        }
+
+#if SUPPORTS_RUNTIME_INTRINSICS
+        /// <summary>
+        /// AVX-only variant for executing <see cref="TransposeInto(ref Block8x8F)"/>.
+        /// <see href="https://stackoverflow.com/questions/25622745/transpose-an-8x8-float-using-avx-avx2/25627536#25627536"/>
+        /// </summary>
+        [MethodImpl(InliningOptions.ShortMethod)]
+        public void TransposeIntoAvx(ref Block8x8F d)
+        {
+            Vector256<float> r0 = Avx.InsertVector128(
+               Unsafe.As<Vector4, Vector128<float>>(ref this.V0L).ToVector256(),
+               Unsafe.As<Vector4, Vector128<float>>(ref this.V4L),
+               1);
```

Comment:
This doesn't change the assembly that is generated but it might show the intention better as:

```suggestion
            Vector256<float> r0 = Vector256.Create(
                Unsafe.As<Vector4, Vector128<float>>(ref this.V0L), 
                Unsafe.As<Vector4, Vector128<float>>(ref this.V4L)
            );
```

Reply:
Can anyone explain the difference in the precision of the output instruction here?

`vmovups` to `vmovupd` at `L0006`

[Original](https://sharplab.io/#v2:C4LghgzgtgPgAgJgIwFgBQcAMACOSB0AcgK5QCmATgJYDGEA3OlrgQErEB2wV5+AwgHsoAByoAbSgGVKAN1pkGTHHnzsuPMvgCSXahwi1FGZW07deO4HoN18ADQAcANkZp02D7gDM2CFeI0wNgAQmICNADWDgAeDgBi7p4A3omennA+NAL6QVRc2JJUAF5k2AC82E4ALK6pad7YAGpkgQIUVU2YADKu9ek+za3tnay1aH0eGU0twG0djUg9dWlTg7PDC6Poy/3TQ/MIS+MTqzNzTQhbbsd9p/tNXkcTDWvnjV5XO5MDZxtVTycfvdGlVPjd6nd1vMAKwA25AqFNaFg56Qt5OOEQhHolGAvaIxoAdkxK2xG0JozSX2w1IA2gBZMjAAAWAgAJloRGIABSMlnsznCMQAeWE3GyEHwAEEAOYyigKAwyMg6MR5PIygCUAF1qVM4B0ACoUMD6YQCCAqrgCKUyaLchUAMxCYUiMXi2DZmupKXBz1eFAQ0KcAB5HWEwMAAHzYCg4Cq26LafSUYABpAIBzc6kTACq+jAjs0UogIYDVQANPi2hmHGGI9Gow6yM6WVRJY1upr8IaBAGg05uZqKzm+vmIIXi6Xy1X05n6wJI1Gm07sG2O/9h6P6khNb0JgBfal1A9AA=)

```
Block8x8F.TransposeIntoAvx(Block8x8F ByRef)
    L0000: sub esp, 0x20
    L0003: vzeroupper
    L0006: vmovups xmm0, [ecx]
    L000a: add ecx, 0x80
    L0010: vmovupd xmm1, [ecx]
    L0014: vinsertf128 ymm0, ymm0, xmm1, 1
    L001a: vmovupd [esp], ymm0
    L001f: vzeroupper
    L0022: add esp, 0x20
    L0025: ret
```

[Suggested Change](https://sharplab.io/#v2:C4LghgzgtgPgAgBgARwIwDoByBXKBTAJwEsBjCAbgFgAoRFDAJWwDtgj90BhAeygAciAG0IBlQgDdSeCjTpp0TVuzzoAkq2LMIpGbWTzFbDuuCbtZdAA0AHADYq1GkVaFmYQUj5gCbd0gim2CTASABCgtwkANbWAB7WAGI0SClIAN7JqVkA9LlIADwQuFDeAJ4AfJlZKbnZSACCSMy4AEaESNwAZv4k7t5IJNx4nZ2kRHisEEjOSGAFEHh4AwTDALwAROGRMfEJ69mV1NWptQXZRVAlBBVV1XAAzAPcWiHOISJEAF5Lq0i2ACwOW5ZB5IABqeGC3AI/3BCAAMg5jilQRCoTC4QwgUdkajIcBobCwahEcDUnj0UTUFiaGSUY80QSMWCAEyknHHClMoksmmODl3Bn4wng+7s5EoIWU0V8umS8HC5n/cW4qXc8H/WUCkFqkVggCsKs5uuZ+q1Eq5etsRsFCulYNs5tVdvVYIA7DadS69W6+VlbqC4LCACoEMBaPjcBYmbj1cSxAAUK26W2icUSSAAJgBKOUZbXIuVZRnQln62z5ToRMDAcpIAjIX4lghl2xcFY1vAJovIgCqWjAnRU9Qg+Wb/wANN6CKgWdZK9Xa+Uk8MkMAABZECDoMEI7NTnvHfsQQfD0fjqfN2fzqvcGvlZfJteb7dg5W5gvHbNI5EAXzpv5AA==)

```
Block8x8F.TransposeIntoAvx(Block8x8F ByRef)
    L0000: sub esp, 0x20
    L0003: vzeroupper
    L0006: vmovupd xmm0, [ecx]
    L000a: add ecx, 0x80
    L0010: vmovupd xmm1, [ecx]
    L0014: vinsertf128 ymm0, ymm0, xmm1, 1
    L001a: vmovupd [esp], ymm0
    L001f: vzeroupper
    L0022: add esp, 0x20
    L0025: ret
```

Reply:
Could be error but I spotted a slight slow down in the change. Keeping as-is for now.